src/Compilers/CSharp/Portable/Symbols/Metadata/PE/SymbolFactory.cs

```diff
@@ -57,6 +57,11 @@ internal override bool IsVolatileModifierType(PEModuleSymbol moduleSymbol, TypeS
             return type.SpecialType == SpecialType.System_Runtime_CompilerServices_IsVolatile;
         }
 
+        internal override bool IsConstModifierType(PEModuleSymbol moduleSymbol, TypeSymbol type)
+        {
+            return type.ToDisplayString() == WellKnownTypes.GetMetadataName(WellKnownType.System_Runtime_CompilerServices_IsConst);
+        }
```

Comment:
From what I tracked down, it ends up doing a lookup in `NamespaceOrTypeSymbol` based on string comparison as well. Will separate the comparison to namespace/type pair to improve performance.

---
In reply to: [117866143](https://github.com/dotnet/roslyn/pull/19658#discussion_r117866143) [](ancestors = 117866143)